runtime/vm/resolvesupport.cpp

```diff
@@ -122,6 +123,27 @@ packageAccessIsLegal(J9VMThread *currentThread, J9Class *targetClass, j9object_t
 	return legal;
 }
 
+BOOLEAN
+requirePackageAccessCheck(J9VMThread *vmThread, J9ClassLoader *srcClassLoader, J9Module *srcModule, J9Class *targetClass)
+{
+	BOOLEAN checkFlag = TRUE;
+	if (J2SE_VERSION(vmThread->javaVM) >= J2SE_19) {
+		if (srcModule == targetClass->module) {
+			if (NULL != srcModule) {
+				/* same named module */
+				checkFlag = FALSE;
+			} else if (J9VMJAVALANGCLASSLOADER_UNNAMEDMODULE(vmThread, srcClassLoader->classLoaderObject)
```

Comment:
doing `srcClassLoader == targetClass->classLoader` might be cheaper. If the classloaders are the same, they will have the same unnamedModule

Reply:
Agree and fixed.